nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1340 @@
+        ----------
+        index : int
+            Indexes the row/column of interest
+
+        Returns
+        -------
+        tuple with 2 elements
+        - unicode name of the row/column
+        - dictionary with the label table
+        - dictionary with the element metadata
+        """
+        return self.name[index], self.label[index], self.meta[index]
+
+
+class Series(Axis):
+    """
+    Along this axis of the CIFTI vector/matrix the rows/columns increase monotonously in time
+
+    This Axis describes the time point of each row/column.
+
+    Attributes
+    ----------
+    start : float
+        starting time point
+    step :  float
+        sampling time (TR)
+    size : int
+        number of time points
+    """
+    size = None
+
+    def __init__(self, start, step, size, unit="SECOND"):
+        """
+        Creates a new Series axis
+
+        Parameters
+        ----------
+        start : float
+            Time of the first datapoint
+        step : float
+            Step size between data points
+        size : int
+            Number of data points
+        unit : str
+            Unit of the step size (one of 'second', 'hertz', 'meter', or 'radian')
+        """
+        self.unit = unit
+        self.start = start
+        self.step = step
+        self.size = size
+
+    @property
+    def time(self):
+        return np.arange(self.size) * self.step + self.start
+
+    @classmethod
+    def from_mapping(cls, mim):
+        """
+        Creates a new Series axis based on a CIFTI dataset
+
+        Parameters
+        ----------
+        mim : cifti2.Cifti2MatrixIndicesMap
+
+        Returns
+        -------
+        Series
+        """
+        start = mim.series_start * 10 ** mim.series_exponent
+        step = mim.series_step * 10 ** mim.series_exponent
+        return cls(start, step, mim.number_of_series_points, mim.series_unit)
+
+    def to_mapping(self, dim):
+        """
+        Converts the Series to a MatrixIndicesMap for storage in CIFTI format
+
+        Parameters
+        ----------
+        dim : int
+            which dimension of the CIFTI vector/matrix is described by this dataset (zero-based)
+
+        Returns
+        -------
+        cifti2.Cifti2MatrixIndicesMap
+        """
+        mim = cifti2.Cifti2MatrixIndicesMap([dim], 'CIFTI_INDEX_TYPE_SERIES')
+        mim.series_exponent = 0
+        mim.series_start = self.start
+        mim.series_step = self.step
+        mim.number_of_series_points = self.size
+        mim.series_unit = self.unit
+        return mim
+
+    _unit = None
+
+    @property
+    def unit(self):
+        return self._unit
+
+    @unit.setter
+    def unit(self, value):
+        if value.upper() not in ("SECOND", "HERTZ", "METER", "RADIAN"):
+            raise ValueError("Series unit should be one of " +
+                             "('second', 'hertz', 'meter', or 'radian'")
+        self._unit = value.upper()
+
+    def extend(self, other_axis):
```

Comment:
The expectation with a method named `extend` is in-place extension, as in lists. If possible, I'd prefer not to violate these expectations. Could this implementation simply be renamed to `__add__`?